rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizationFileReader.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.manifests.kustomize;
+
+import com.google.common.collect.ImmutableList;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import com.netflix.spinnaker.rosco.services.ClouddriverService;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.representer.Representer;
+import retrofit.client.Response;
+
+@Component
+@Slf4j
+public class KustomizationFileReader {
+  private final ClouddriverService clouddriverService;
+  private final RetrySupport retrySupport = new RetrySupport();
+  private static final List<String> KUSTOMIZATION_FILENAMES =
+      ImmutableList.of("kustomization.yaml", "kustomization.yml", "kustomization");
+
+  public KustomizationFileReader(ClouddriverService clouddriverService) {
+    this.clouddriverService = clouddriverService;
+  }
+
+  public Kustomization getKustomization(Artifact artifact, String possibleName) {
+    Path artifactPath = Paths.get(artifact.getReference());
```

Comment:
I'm a bit confused by the logic in this function...where is the `possibleName` coming from?  It looks like we're trying all of the `KUSTOMIZATION_FILENAMES` but using an input String as a hint for what to try first?

Reply:
Correct. A Kustomization file can have 3 names, `.yml`, `.yaml` and one w/o an extension. Initially, the file name comes from the artifact that you supply as an input. We're making the assumption that if you use, say `kustomization.yml`, you're likely to use `kustomization.yml` for your subsequent kustomizations. Since we have no way of knowing that the actual file name is before we try to fetch it we try to be smart and make a best guess before falling back to the other options.